From 07a237491893f4c0ac96cce59c58eb6010b8a61d Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Fri, 13 Feb 2015 01:02:03 +0800
Subject: [PATCH 0095/1691] MLK-10257-03 arm: imx7: Add cpufreq platform device

commit 07a237491893f4c0ac96cce59c58eb6010b8a61d from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Add cpufreq platform device support for DVFS.

Signed-off-by: Bai Ping <b51503@freescale.com>
(cherry picked from commit 27642021d28555c37aab5e2b75214d8c383f668d)
---
 arch/arm/mach-imx/mach-imx7d.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/mach-imx/mach-imx7d.c b/arch/arm/mach-imx/mach-imx7d.c
index 8634389..0923dab 100644
--- a/arch/arm/mach-imx/mach-imx7d.c
+++ b/arch/arm/mach-imx/mach-imx7d.c
@@ -11,6 +11,7 @@
 #include <linux/mfd/syscon.h>
 #include <linux/mfd/syscon/imx7-iomuxc-gpr.h>
 #include <linux/of_platform.h>
+#include <linux/pm_opp.h>
 #include <linux/phy.h>
 #include <linux/regmap.h>
 
@@ -110,6 +111,13 @@ static void __init imx7d_init_irq(void)
 	irqchip_init();
 }
 
+static void __init imx7d_init_late(void)
+{
+	if (IS_ENABLED(CONFIG_ARM_IMX7D_CPUFREQ)) {
+		platform_device_register_simple("imx7d-cpufreq", -1, NULL, 0);
+	}
+}
+
 static const char *const imx7d_dt_compat[] __initconst = {
 	"fsl,imx7d",
 	"fsl,imx7s",
@@ -126,5 +134,6 @@ DT_MACHINE_START(IMX7D, "Freescale i.MX7 Dual (Device Tree)")
 	.map_io	= imx7d_map_io,
 	.init_irq	= imx7d_init_irq,
 	.init_machine	= imx7d_init_machine,
+	.init_late	= imx7d_init_late,
 	.dt_compat	= imx7d_dt_compat,
 MACHINE_END
-- 
1.9.1

